src/relay/libp2p_relay_server.erl

```diff
@@ -66,6 +68,18 @@ connection_lost(Swarm) ->
             Error
     end.
 
+-spec stop(pid()) -> ok | {error, any()}.
+stop(Swarm) ->
+    case get_relay_server(Swarm) of
+        {ok, Pid} ->
+            %% it's a permanant worker, so just stop it and let it restart
+            %% without having relay() called on it
```

Comment:
Why not make relay server a transient instead?

Reply:
Because I didn't want to do too much surgery on this without knowing if it will help, I guess

Reply:
It currently always runs, and we just conditionally tell it to start maintaining a relay address for us. I just stop it to return it to the initial state after it restarts.